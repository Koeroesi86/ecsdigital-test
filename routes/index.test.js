const routes = require('./index');

jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({ data: {} })),
}));

jest.mock('./cars', () => ({
  add: jest.fn(() => Promise.resolve()),
  all: jest.fn(() => Promise.resolve([])),
  get: jest.fn(() => Promise.resolve({})),
  update: jest.fn(() => Promise.resolve()),
  remove: jest.fn(() => Promise.resolve()),
}));

/** Mock express */
const app = {
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
  delete: jest.fn(),
};

describe('The routes', () => {
  it('should register routes', () => {
    routes(app);

    expect(app.get.mock.calls).toMatchSnapshot();
    expect(app.post.mock.calls).toMatchSnapshot();
    expect(app.put.mock.calls).toMatchSnapshot();
    expect(app.delete.mock.calls).toMatchSnapshot();
  });
});

