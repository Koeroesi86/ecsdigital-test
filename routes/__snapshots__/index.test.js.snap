// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The routes should register routes 1`] = `
Array [
  Array [
    "/cars",
    [Function],
  ],
  Array [
    "/cars/:id",
    [Function],
  ],
]
`;

exports[`The routes should register routes 2`] = `
Array [
  Array [
    "/cars",
    [Function],
  ],
]
`;

exports[`The routes should register routes 3`] = `
Array [
  Array [
    "/cars/:id",
    [Function],
  ],
]
`;

exports[`The routes should register routes 4`] = `
Array [
  Array [
    "/cars/:id",
    [Function],
  ],
]
`;
